src/test/java/org/apache/commons/text/similarity/SorensenDiceSimilarityTest.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+package org.apache.commons.text.similarity;
+
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit tests for {@link SorensenDicesSimilarity}.
+ */
+public class SorensenDiceSimilarityTest {
+
+    private static SorensenDiceSimilarity similarity;
+
+    @BeforeAll
+    public static void setUp() {
+        similarity = new SorensenDiceSimilarity();
+    }
+
+    @Test
+    public void test() {
+        assertEquals(0.25d, similarity.apply("night", "nacht"));
+    }
+
+    @Test
+    public void testGetSorensenDicesSimilarity_StringString() {
+
+        assertEquals(1d, similarity.apply("", ""));
+        assertEquals(0d, similarity.apply("", "a"));
+        assertEquals(0d, similarity.apply("a", ""));
+        assertEquals(1d, similarity.apply("a", "a"));
+        assertEquals(0d, similarity.apply("a", "b"));
+        assertEquals(1.0d, similarity.apply("foo", "foo"));
+        assertEquals(0.8d, similarity.apply("foo", "foo "));
+        assertEquals(0.4d, similarity.apply("frog", "fog"));
+        assertEquals(0.0d, similarity.apply("fly", "ant"));
+        assertEquals(0.0d, similarity.apply("elephant", "hippo"));
+        assertEquals(0.0d, similarity.apply("hippo", "elephant"));
+        assertEquals(0.0d, similarity.apply("hippo", "zzzzzzzz"));
+        assertEquals(0.5d, similarity.apply("hello", "hallo"));
+        assertEquals(0.7d, round(similarity.apply("ABC Corporation", "ABC Corp"), 1));
+        assertEquals(0.7d, round(similarity.apply("D N H Enterprises Inc", "D &amp; H Enterprises, Inc."), 1));
+        assertEquals(0.8d,
+                round(similarity.apply("My Gym Children's Fitness Center", "My Gym. Childrens Fitness"), 1));
+        assertEquals(0.7d, round(similarity.apply("PENNSYLVANIA", "PENNCISYLVNIA"), 1));
+        assertEquals(0.9d, round(similarity.apply("/opt/software1", "/opt/software2"), 1));
+        assertEquals(0.6d, round(similarity.apply("aaabcd", "aaacdb"), 1));
+        assertEquals(0.6d, round(similarity.apply("John Horn", "John Hopkins"), 1));
+
+    }
+
+    @Test
+    public void testGetSorensenDicesSimilarity_NullNull() {
+        assertThatIllegalArgumentException().isThrownBy(() -> {
+            similarity.apply(null, null);
+        });
+    }
+
+    @Test
+    public void testGetSorensenDicesSimilarity_StringNull() {
+        assertThatIllegalArgumentException().isThrownBy(() -> {
+            similarity.apply(" ", null);
+        });
+    }
+
+    @Test
+    public void testGetSorensenDicesSimilarity_NullString() {
+        assertThatIllegalArgumentException().isThrownBy(() -> {
+            similarity.apply(null, "clear");
+        });
+    }
+
+    public static double round(double value, int precision) {
```

Comment:
No need to be public. Why are you rounding anyway? A test like this can be very precise.